code/modules/mob/living/simple_animal/hostile/pulse_demon/pulsedemon_powers.dm

```diff
@@ -630,3 +608,35 @@
 		explosion(get_turf(M), -1, 1, 2, 3, whodunnit = user) //C4 Radius + 1 Dest for the machine
 		qdel(M)
 	..()
+
+/datum/action/pd_toggle_drain
+	name = "Toggle power drain"
+	desc = "Toggles the draining of power while in an APC, battery or cable"
+	icon_icon = 'icons/mob/screen_spells.dmi'
+	button_icon_state = "pd_toggle"
+
+/datum/action/pd_toggle_drain/Trigger()
+	if(istype(owner,/mob/living/simple_animal/hostile/pulse_demon))
```

Comment:
This was probably written before the ispulsedemon define huh?

Reply:
no i just forgot, will fix it